oaknational/Oak-Web-Application
# Feat/2023 downloads LESQ-220

## Description

- Adapts hooks to take view type prop to determine which endpoint to check and download resources.

## Issue(s)

Fixes #

## How to test

This unit has a working download for worksheet questions
/beta/teachers-2023/programmes/geography-secondary-ks4/units/test-unit/lessons/lesson-1-in-test-unit/downloads
When the page loads an existence check (fetch) is made to the new downloads api a validation error is returned.(see ticket below LESQ-272)
I
When you click downloads with just the worksheet selected,  worksheet is downloaded.
```
[
    "Invalid selection, the following resources don't exist for this lesson: presentation, intro-quiz-questions, exit-quiz-questions"
]
```

**Extra tickets needed**

Assets can only be downloaded from the downloads api if they have been ‘published’, we are currently looking at the id of an asset to determine if it exists and can be downloaded. We need to also check the status. The lesson below has a ‘published’ worksheet pdf, the other assets are ‘new’ but the MV thinks they exist and can be downloaded.
ticket - https://www.notion.so/oaknationalacademy/I-want-the-downloads-page-to-only-show-downloads-for-resources-that-are-published-2cdf611abf0944aba0d7205bff83e2a9


Noticed the download all button is not yet connected to the MV - ticket here https://www.notion.so/oaknationalacademy/I-want-a-lesson-to-know-if-it-has-downloadable-resources-379fae4b803d44b78ecbcf6689e1cf49


1. Go to https://deploy-preview-1834--oak-web-application.netlify.thenational.academy
2. Click on _______
3. You should see _______

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
